codec-http2/src/main/java/io/netty/handler/codec/http2/HpackDynamicTable.java

```diff
@@ -83,7 +83,8 @@ public long capacity() {
      */
     public HpackHeaderField getEntry(int index) {
         if (index <= 0 || index > length()) {
-            throw new IndexOutOfBoundsException();
+            throw new IndexOutOfBoundsException("expected: 0 < index(" + index + ") <= length("
+                    + length() + ')');
```

Comment:
JDK9 introduced the `Objects.checkIndex​(int index, int length)` method. Consider using the wording consistent with that error message:

https://github.com/AdoptOpenJDK/openjdk-jdk11/blob/19fb8f93c59dfd791f62d41f332db9e306bc1422/src/java.base/share/classes/jdk/internal/util/Preconditions.java#L188-L189